src/main/java/lv/ctco/javaschool/goal/control/TagParser.java

```diff
@@ -0,0 +1,54 @@
+package lv.ctco.javaschool.goal.control;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TagParser {
+    public static boolean isEqualSets(Set<?> set1, Set<?> set2) {
+        if (set1 == null && set2 == null) {
+            return true;
+        }
+        if (set1 == null || set2 == null) {
+            return false;
+        }
+        if (set1.size() != set2.size()) {
+            return false;
+        }
+        return set1.containsAll(set2);
+    }
+
+    public static boolean isEqualLists(List<?> list1, List<?> list2) {
```

Comment:
This implementation is re-inventing the wheel.
1) Have a look at `AbstractList.equals()` which already covers some of these checks.
2) I believe the optimal solution would be `return Objects.equals(list1, list2);`.